flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointsCleaner.java

```diff
@@ -107,6 +144,7 @@ private void maybeCompleteCloseUnsafe() {
         if (numberOfCheckpointsToClean == 0 && cleanUpFuture != null) {
             cleanUpFuture.complete(null);
         }
+        subsumedCheckpoints.clear();
```

Comment:
Shouldn't this statement be placed under `if`? 
Otherwise, it will clear `subsumedCheckpoints` after every checkpoint cleanup?

Reply:
Right, moved it into `if`